### ISSUE TITLE
Prevent duplicated targets

### ISSUE THREAD

**Issue by xakrume** [warmans/prometheus-aggregate-exporter#23]
Is it possible to check existing/registered targets and prevent re-registration?
```bash
dev@test ~ $ /usr/local/bin/prometheus-aggregate-exporter -server.bind="127.0.0.1:8080" -targets="test1=http://localhost:10004/metrics,test2=http://localhost:7116/metrics" -targets.dynamic.registration="true" -targets.cache.path="/tmp/targets.conf" -verbose
2021/07/28 12:33:09 Dynamic target registration enabled
2021/07/28 12:33:09 Recovered target test3=http://localhost:7109/metrics from cache file
2021/07/28 12:33:09 Recovered target test4=http://localhost:10014/metrics from cache file
2021/07/28 12:33:09 Using targets cache file /tmp/targets.conf
2021/07/28 12:33:09 Starting server on 127.0.0.1:8080 with targets:
2021/07/28 12:33:09   - test1=http://localhost:10004/metrics
2021/07/28 12:33:09   - test2=http://localhost:7116/metrics
2021/07/28 12:33:09   - test3=http://localhost:7109/metrics
2021/07/28 12:33:09   - test4=http://localhost:10014/metrics
....
```
Register already registered target:
```bash
dev@test ~ $ curl -v 'http://127.0.0.1:8080/register?name=test3&address=localhost:7109/metrics'
dev@test ~ $ curl -v 'http://127.0.0.1:8080/register?name=test3&address=localhost:7109/metrics'
```

`prometheus-aggregate-exporter` verbose output:
```
...
2021/07/28 12:33:09   - test3=http://localhost:7109/metrics
2021/07/28 12:33:09   - test4=http://localhost:10014/metrics
2021/07/28 12:33:21 Registered target http://localhost:7109/metrics with name test3
2021/07/28 12:33:57 Registered target http://localhost:7109/metrics with name test3
```

Get stats from `prometheus-aggregate-exporter` with duplicated targets and as a result with
duplicated metrics, but not duplicated values, perhaps there several requests for duplicated targets and the values of which have been placed in the corresponding duplicated targets:

There may have been a duplication of objectives, and as a result several queries have been made, the values of which have been placed in the corresponding

```bash
dev@test ~ $ curl 127.0.0.1:8080/metrics
# TYPE go_memstats_heap_objects gauge
go_memstats_heap_objects{ae_source="test1"} 76215
go_memstats_heap_objects{ae_source="test4"} 33068
go_memstats_heap_objects{ae_source="test3"} 48086
go_memstats_heap_objects{ae_source="test3"} 47893
go_memstats_heap_objects{ae_source="test3"} 48240
go_memstats_heap_objects{ae_source="test2"} 126888
...
```

`prometheus-aggregate-exporter` verbose output:
```
...
2021/07/28 12:33:21 Registered target http://localhost:7109/metrics with name test3
2021/07/28 12:33:57 Registered target http://localhost:7109/metrics with name test3
2021/07/28 12:46:31 OK: test1=http://localhost:10004/metrics was refreshed in 0.004 seconds
2021/07/28 12:46:31 OK: test4=http://localhost:10014/metrics was refreshed in 0.005 seconds
2021/07/28 12:46:31 OK: test3=http://localhost:7109/metrics was refreshed in 0.007 seconds
2021/07/28 12:46:31 OK: test3=http://localhost:7109/metrics was refreshed in 0.007 seconds
2021/07/28 12:46:31 OK: test3=http://localhost:7109/metrics was refreshed in 0.007 seconds
2021/07/28 12:46:31 OK: test2=http://localhost:7116/metrics was refreshed in 0.011 seconds
```

**Comment by xakrume** [comment]
It would be good to avoid duplicated targets in the cache file and in the running service with `dynamic target registration` enabled

**Comment by warmans** [comment]
This should be fixed in the v3.0.0 tag. Docker hub no longer does automated builds so until I sort out Github Actions, I've had to push the image manually. Hopefully it works 🤷‍♂️
